src/main/java/net/sf/mzmine/modules/peaklistmethods/identification/sirius/PeakListIdentificationTask.java

```diff
@@ -84,10 +85,18 @@
     fingerCandidates =
         parameters.getParameter(PeakListIdentificationParameters.CANDIDATES_FINGERID).getValue();
 
+    String massListName = parameters.getParameter(PeakListIdentificationParameters.MASS_LIST).getValue();
+    List<String> massLists = MassListComponent.getMassListNames();
+
     if (timer <= 0 || siriusCandidates <= 0 || fingerCandidates <= 0 || threadsAmount <= 0) {
       MZmineCore.getDesktop().displayErrorMessage(MZmineCore.getDesktop().getMainWindow(),
           "Sirius parameters can't be negative");
       cancel();
+    } else if (!massLists.contains(massListName)) {
```

Comment:
This is still not right - you are checking if the massListName exists in the list of all mass list names - but actually you need to check for every scan separately. Each scan that you are loading must have a mass list of that name.

Reply:
Do you mean each scan that is used to be identified by Sirius?
But it is probably that some of the scans don't have it, for example even ATP molecule, but it is still identified.
Here is an example of logging information for ATP:
```
[23:59:04|INFO|WorkerThread]: Starting processing of task Peak identification of 508.005 using Sirius module
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1065. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1087. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1109. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1097. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1121. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1145. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1067. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1089. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1111. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1081. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1103. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1093. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1117. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1139. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1085. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1109. Row id = 389
[23:59:04|FINE|SpectrumScanner]: [masses MS2] mass list does not exist in a scan = 1141. Row id = 389
[23:59:07|FINE|ConstraintsGenerator]: ConstraintsGenerator started processing
[23:59:07|FINE|ConstraintsGenerator]: ConstraintsGenerator finished
[23:59:07|INFO|NativeLibraryLoader]: Started loading libraries from glpk-4.60 folder
[23:59:07|INFO|NativeLibraryLoader]: OS type = windows and OS arch = 64
[23:59:07|INFO|NativeLibraryLoader]: Loading libraries from JAR
[23:59:07|INFO|NativeLibraryLoader]: Successfully loaded glpk_4_60 library
[23:59:07|INFO|NativeLibraryLoader]: Successfully loaded glpk_4_60_java library
```

Reply:
We agreed on the approach that the mass list name should be the same for MS1 and MS2 scans. In the sample project, the mass list name is different, that why you should return an error. After running mass detection on all scans, it will work fine.